src/main/java/org/opensearch/performanceanalyzer/commons/os/ThreadSched.java

```diff
@@ -37,6 +39,27 @@ public static class SchedMetrics {
             this.contextSwitchRate = contextSwitchRate;
         }
 
+        @Override
+        public int hashCode() {
```

Comment:
Why are those functions needed? Also if these functions will only be used in the unit tests, let's try adding some similar util functions in the test itself instead of adding new functions for the class.

Reply:
These functions are used to implement equality, which is used to verify that the `setSchedMetric` is called with a `ThreadSched.SchedMetrics` that has the correct values.

i.e.

```java
        verify(linuxSchedMetricsGenerator)
                .setSchedMetric("3", new ThreadSched.SchedMetrics(0.002, 0.002, 100.0));
```
at https://github.com/DesmondWillowbrook/performance-analyzer-commons/blob/threadsched-test/src/test/java/org/opensearch/performanceanalyzer/commons/os/ThreadSchedTests.java#L123-L124

Would you prefer something that uses the [Reflection](https://www.oracle.com/technical-resources/articles/java/javareflection.html) API instead?

Reply:
@ansjcy Does that explain why I implemented equality?